src/Serialization/containers.jl

```diff
@@ -29,13 +40,23 @@ end
 
 # deserialize without specific content type
 function load_internal(s::DeserializerState, ::Type{Vector}, dict::Dict)
+    if haskey(dict, :entry_type)
+        T = decodeType(dict[:entry_type])
+        
```

Comment:
Should we be paranoid here and insert a check?
```suggestion
        @assert is_basic_serialization_type(T)

```

Reply:
this will still work if ```is_basic_serialization_type(T) == false```
and I we can also in the future extend this type of serialization to other non basic serialization types?
the only way it can fail is if someone goes into file and entry_type that isn't correct for the type of data it is